src/routes/bounce_details_page.jsx

```diff
@@ -83,8 +86,38 @@ export default class BounceDetailsPage extends React.Component {
     });
   }
 
+  onEditRuleCommit(e) {
+    const { updatedRule } = this.state;
+    const { id, value } = e.currentTarget;
+    if (value.length === 0) {
```

Comment:
Hmm we're doing a lot of double setState in the same edit function which will lead to two renders and there is a lot of duplicate logic inside each of the handlers that is copied everywhere. We can extract out the logic to be something like ```const isCommitValid = validateCommit(value)``` and then later do one setState like ``` this.setState({ updatedRule: { ...updatedRule, [id]: value }, isCommitEmpty: isCommitValid, isCommitDisabled: isCommitDisabled })```. Now that I think about it we probably don't event need the isCommitDisabled state since it's computed from the isCommitEmpty state variable. We can probably get away with only isCommitValid/Empty or something like that and not duplicate the code everywhere